inst/table.m

```diff
@@ -500,9 +500,21 @@ function prettyprint (this)
       out = this;
     endfunction
     
-    function out = size_equal
+    ## -*- texinfo -*-
+    ## @node table.size_equal
+    ## @deftypefn {Method} {@var{out} =} size_equal (@var{varargin})
+    ##
+    ## True if the dimensions of all arguments agree.
+    ##
+    ## @end deftypefn
+    function out = size_equal (varargin)
       %SIZE_EQUAL True if the dimensions of all arguments agree.
-      error ('table.size_equal: size_equal is not yet implemented for tables');
+      [nr, nv] = size(varargin{1});
+      out = 1;
+      for k = 2:length(varargin)
```

Comment:
Tablicious uses `i` or `i<Thing>` as its conventional loop index variable names. Could this be `i` instead of `k`?

Reply:
Please use `numel` instead of `length`. `length` will work fine here, but it is problematic in general, so I prefer to avoid it entirely.

Reply:
Done. Would love to hear more about how `length` is problematic vs. `numel`, as I am definitely in the habit of using the former.

Reply:
The problem is that `length` is defined to return the size along the _longest_ dimension of an array. That is fine for use with an input that you are _sure_ is always going to be a vector, but when you deal with 2-D or higher arrays, that introduces subtle edge cases where if you get an array of unusual size or dimensionality, it doesn't raise an error, but might return the size along the wrong dimension! Which can silently produce incorrect results.

`numel` always returns exactly the total number of elements in an array, which is usually what you want instead, especially in the case where you're iterating over something.

So rather than having to worry about making sure I know inputs are vectors when I want to use `length`, I just use the rule "never use `length`" and I never have to worry about this problem case.

Reply:
Interesting ... I think I just saw from the MATLAB documentation that `length(A)` is equivalent to `max(size(A))` and `numel(A)` to `prod(size(A))` and figured `length` might be a smidge faster, at least when I know I'm dealing with a vector. Thanks.

Reply:
Nah... all those calculations are trivial (because all the size information for a Matlab or Octave array is stored in a little bitty header section for each array), so the cost of actually performing them will be swamped by the overhead of the function call itself. No performance considerations here.